models/devices.go

```diff
@@ -183,6 +212,96 @@ type OmeManagementProfileData struct {
 
 // ################### tfsdk converters
 
+// NewDeviceItemModel - converts Device to DeviceItemModel
+func NewDeviceItemModel(input Device) DeviceItemModel {
+	ips := make([]string, 0)
+	for _, m := range input.DeviceManagement {
+		ips = append(ips, m.NetworkAddress.String())
+	}
+	return DeviceItemModel{
+		ID:         types.Int64Value(input.ID),
+		ServiceTag: types.StringValue(input.DeviceServiceTag),
+		IPs:        stringListValue(ips),
+	}
+}
+
+// NewDevicesResModel - converts list of json devices to DevicesResModel
+func NewDevicesResModel(input []Device) (DevicesResModel, diag.Diagnostics) {
+	devs := make([]DeviceItemModel, 0)
+	for _, v := range input {
+		devs = append(devs, NewDeviceItemModel(v))
+	}
+	tfdevs, dgs := objListValue(DeviceItemModel{}.getType(), devs)
+	return DevicesResModel{
+		ID:      types.StringValue("dummy"),
+		Devices: tfdevs,
+	}, dgs
+}
+
+// NewDevicesResModelFromID - converts a string id into a terraform plan
+func NewDevicesResModelFromID(id string) (types.List, diag.Diagnostics) {
+	var dgs diag.Diagnostics
+	if id == "" {
+		return types.ListUnknown(types.ObjectType{
+			AttrTypes: DeviceItemModel{}.getType(),
+		}), dgs
+	}
+	var plan types.List
+	vals := strings.Split(id, ":")
+	var did, didst string
+	if l := len(vals); l == 1 {
+		did = "id"
+		didst = id
+	} else if l > 2 {
+		dgs.AddError(
+			"Wrong id format.",
```

Comment:
ID

Reply:
You mean the "id" string in the diagnostics? Is there any particular reason for this request?
The reason I am asking is that, even though Go conventionally uses "ID" in most cases, the user facing side of terraform almost always goes for "id". This is seen in the required "id" attribute of  all resources as well as in the import block.

Reply:
shall we mention this inside double quotes?
`Wrong \"id\" format.`